crates/resolver/src/config.rs

```diff
@@ -819,6 +854,65 @@ impl<'a> ServerGroup<'a> {
     }
 }
 
+/// This is an alternate serialization function for a [`Duration`] that emits a single number,
+/// representing the number of seconds, instead of a struct with `secs` and `nanos` fields.
+#[cfg(feature = "serde")]
+fn duration_serialize<S>(duration: &Duration, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
Can you inline the bounds for all of these, `duration_serialize<S: Serializer>(..)`?